src/provider/AuthProvider.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react';
+import { AuthContext } from 'context/AuthContext';
+import firebase from 'firebase/compat/app';
+import { auth } from 'config/firebase';
+
+export const AuthProvider: React.FC = ({ children }) => {
```

Comment:
React.FC isn't a good approach. A better way is to create an Interface
https://stackoverflow.com/questions/59988667/typescript-react-fcprops-confusion

I suggest:
```js
interface Props {
    children: React.ReactNode
}

export const AuthProvider = ({ children }: Props) => {
```

Reply:
fixed